sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3312,6 +3313,22 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
         """.stripMargin).collect()
     }
   }
+
+  test("SPARK-29186: SubqueryAlias name value should not be null in json") {
```

Comment:
Hi, @viirya . 
Instead of adding this E2E test, could you add a test case to `TreeNodeSuite.scala`?
There is `test("toJSON")` and more there.

Reply:
Ah, thanks. Looks TreeNodeSuite is a proper place for the test.